server/consumer.go

```diff
@@ -5182,8 +5182,9 @@ func (o *consumer) stopWithFlags(dflag, sdflag, doSignal, advisory bool) error {
 
 		// dflag does not necessarily mean that the consumer is being deleted,
 		// just that the consumer node is being removed from this peer, so we
-		// send delete advisories only if we are no longer assigned at the meta layer.
-		if !isConsumerAssigned && advisory {
+		// send delete advisories only if we are no longer assigned at the meta layer,
+		// or we are not clustered.
+		if (!isConsumerAssigned || !o.js.isClustered()) && advisory {
```

Comment:
Do we need to capture o.js when the consumer lock is held?

Actually capture it up when we check the dflag, just set isClustered := o.js.isClustered() and use it to determine that we are assigned there and leave above the same logic.

Reply:
```go
func (o *consumer) stopWithFlags(dflag, sdflag, doSignal, advisory bool) error {
	// If dflag is true determine if we are still assigned.
	var isConsumerAssigned bool
	if dflag {
		o.mu.RLock()
		acc, stream, consumer := o.acc, o.stream, o.name
		isClustered := o.js.isClustered()
		o.mu.RUnlock()
		if isClustered {
			// Grab jsa to check assignment.
			var jsa *jsAccount
			if acc != nil {
				// Need lock here to avoid data race.
				acc.mu.RLock()
				jsa = acc.js
				acc.mu.RUnlock()
			}
			if jsa != nil {
				isConsumerAssigned = jsa.consumerAssigned(stream, consumer)
			}
		} else {
			// Always true in non-clustered mode.
			isConsumerAssigned = true
		}
	}
```

Reply:
Ok. I simplified that into setting bool determining if consumer is deleted instead of calculating it each time too.